test/functional/uvm_memory_test.go

```diff
@@ -0,0 +1,56 @@
+package functional
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/Microsoft/hcsshim/osversion"
+	testutilities "github.com/Microsoft/hcsshim/test/functional/utilities"
+)
+
+func TestUVMMemoryUpdateLCOW(t *testing.T) {
+	testutilities.RequiresBuild(t, osversion.RS5)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 40*time.Second)
+	defer cancel()
+
+	opts := uvm.NewDefaultOptionsLCOW(t.Name(), "")
+	opts.MemorySizeInMB = 1024 * 2
+	u := testutilities.CreateLCOWUVMFromOpts(ctx, t, opts)
+	defer u.Close()
+
+	newMemorySize := uint64(opts.MemorySizeInMB/2) * bytesPerMB
+
+	if err := u.UpdateMemory(ctx, newMemorySize); err != nil {
+		t.Fatalf("failed to make call to modify UVM memory size in MB with: %v", err)
+	}
+}
+
+func TestUVMMemoryUpdateWCOW(t *testing.T) {
+	testutilities.RequiresBuild(t, osversion.RS5)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 40*time.Second)
+	defer cancel()
+
+	opts := uvm.NewDefaultOptionsWCOW(t.Name(), "")
+	opts.MemorySizeInMB = 1024 * 2
+
+	u, _, uvmScratchDir := testutilities.CreateWCOWUVMFromOptsWithImage(ctx, t, opts, "mcr.microsoft.com/windows/nanoserver:1909")
```

Comment:
Any reason for the 1909 image? I see the test requires RS5 or higher so just wondering why not just an RS5 or 19H1 image

Reply:
Honestly, these tests don't get run and they depend on docker to get these image layer files, so this is basically just so we have A test that we could run if something happens or we do more in the future. I've been hoping to look into getting these functional tests running and moving away from docker, but until then.

Reply:
Gotcha